bin/i18n/src/i18n/enumerate.clj

```diff
@@ -0,0 +1,127 @@
+(ns i18n.enumerate
+  "Enumerate and create pot file from the backend worktree of metabase."
+  (:require
+   [clojure.java.io :as io]
+   [clojure.spec.alpha :as s]
+   [clojure.string :as str]
+   [grasp.api :as g]
+   [metabuild-common.core :as u])
+  (:import [org.fedorahosted.tennera.jgettext
+            Catalog HeaderFields HeaderUtil HeaderFields Message PoWriter]))
+
+(set! *warn-on-reflection* true)
+
+(def ^:private roots (into [] (map (partial str u/project-root-directory))
+                           ["/src" "/shared/src" "/enterprise/backend/src"]))
+
+(defn- strip-roots
+  [path]
+  (str/replace path
+               (re-pattern (str/join "|" (map #(str "file:" % "/") roots)))
+               ""))
+
+(s/def ::translate (s/and
+                     (complement vector?)
+                     (s/cat :translate-symbol (fn [x]
+                                               (and (symbol? x)
+                                                    (#{"trs" "deferred-trs"
+                                                       "tru" "deferred-tru"}
+                                                      (name x))))
```

Comment:
Maybe I am paranoid, but this will match any namespaced symbol, if I am reading it right. What do you think of restricting the acceptable namespaces to i18n?

Reply:
Yes that's true. The problem is that I don't have full analysis here. If we went full-bore and used `clj-kondo` to get this information we could do that. But these vars are not resolved so I don't have the full information there. I think our code discipline is good enough though with these vars. Note this limitation is present with the current `xgettext` tool as well though.

Reply:
Maybe I was unclear, I meant that we can check the literal symbols themselves for a namespace nil of "i18n". But tbf it's unlikely this will bite us

Reply:
the symbols almost always don't have a namespace at all. `(:require mb.i18n :refer [trs tru])` is a _very_ common pattern. And even more, that would be checking the alias, not the actual namespace if that pattern wasn't common place.

Reply:
I think the change looks good as-is, I am just not sure that I expressed what I meant clearly here, so let me add on:

> The problem is that I don't have full analysis here. ... that would be checking the alias, not the actual namespace

Checking if the namespace is in `#{nil "i18n"}` wouldn't require full anlysis or resolving the vars.

> that would be checking the alias, not the actual namespace

Unless we remove the linting around it, namespace aliases are always 1:1 with the namespaces they reference. So `i18n` as a symbol's namespace prefix will always point to `metabase.util.i18n`. Or am I missing something there?

> (:require mb.i18n :refer [trs tru]) is a very common pattern.

Yep! that's what I mean: the namespace would commonly be `nil`.

Reply:
best of both worlds: 

```clojure
(def translation-vars
  "Vars that are looked for for translations strings"
  #{'metabase.util.i18n/trs
    'metabase.util.i18n/tru
    'metabase.util.i18n/deferred-trs
    'metabase.util.i18n/deferred-tru
    'metabase.shared.util.i18n/tru})

(translation-vars (g/resolve-symbol))
```